app/gitcoinbot/actions.py

```diff
@@ -20,66 +20,150 @@
 import datetime
 import json
 import re
+from functools import wraps
 
 from django.conf import settings
 
 import jwt
 import requests
+from dashboard.models import Bounty
+from dashboard.tokens import tokens
+from gitcoinbot.models import GitcoinBotResponses
 from github.utils import post_issue_comment_reaction
 
 GITCOINBOT_APP_ID = settings.GITCOINBOT_APP_ID
 SECRET_KEYSTRING = settings.SECRET_KEYSTRING
 
 
+MIN_AMOUNT = 0
+CURRENCIES = set(map(lambda currency: currency["name"], tokens))
+
+
+class Bound(object):
+    def __init__(self, *args):
+        self.bounds = args
+
+    def __call__(self, f):
+        wraps(f)
+
+        def wrapped_f(*args, **kwargs):
+            for bound in self.bounds:  # Check if every bound is complained
+                (valid, msg) = bound(*args, **kwargs)
+                if not valid:
+                    return msg  # especify what bound failed and why
+
+            return f(*args, **kwargs)
+
+        return wrapped_f
+
+
+def amount_greater_than_zero(*args, **kwargs):
+    # No negatives and greater than 0
+    comment_text = args[-1]
+    amount = parse_comment_amount(comment_text)
+    tip_currency = parse_comment_currency(comment_text)
+    response =  f'The amount should be greater than {MIN_AMOUNT} {tip_currency}'
+
+    try:
+      f_amount = float(amount)
+      if f_amount > MIN_AMOUNT:
+        return (True, '')
+    except:
```

Comment:
Can you except on either the specific exception occurring here (maybe `ValueError`) or catch `Exception`